compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -901,17 +904,14 @@ pub(crate) unsafe fn codegen(
                     .generic_activity_with_arg("LLVM_module_codegen_emit_obj", &module.name[..]);
 
                 let dwo_out = cgcx.output_filenames.temp_path_dwo(module_name);
-                let dwo_out = match cgcx.split_debuginfo {
```

Comment:
I'm not sure I understand why this change is necessary - the current implementation _is_ a little confused, packed/unpacked should be distinct from split/single, see #89819's re-introduction of `-Zsplit-dwarf-kind` - but that seems unrelated to the change you're making in this pull request.

Reply:
Before, we don't produce the `.dwo` files when it's packed. We rely on `llvm-dwp` to read the paths to `.o` files in the linked executable to read the debuginfo to pack into a single `.dwp`.

The comment above `link_dwarf_object` is said
```
/// Invoke `llvm-dwp` (shipped alongside rustc) to link `dwo` files from Split DWARF into a `dwp`
/// file.
```

So I thought `llvm-dwp` needs to get paths to `.dwo` files, which we weren't producing for `packed`. But I think it's actually reading `.o`. I reverted the changes in this file and gave `llvm-dwp` object files instead and it seems to work

Reply:
Yeah, that should work, after #89819, the packed/unpacked won't change whether split debuginfo is in dwo files or not, that'll be an independent option.